### PR TITLE
Cypress/E2E: Fix "app.role.save.invalid_role.app_error" when patching roles

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
- Fix "app.role.save.invalid_role.app_error" when patching roles by updating default roles
- Move api commands to its own `/api/role`

### CODE REVIEW

File: e2e/cypress/support/api/role.js [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+        'create_post',
+        'manage_private_channel_members',
+        'read_public_channel_groups',
+        'manage_channel_roles',
+        'use_channel_mentions',
+        'use_group_mentions',
+        'read_private_channel_groups',
+        'remove_reaction',
+        'add_reaction',
+        'manage_public_channel_members',
```
Comment (fmunshi) [inline e2e/cypress/support/api/role.js:22]: @saturninoabril can we add `convert_public_channel_to_private` & `convert_private_channel_to_public` here (for team admin role) to match the server defaults

Comment (fmunshi) [reply e2e/cypress/support/api/role.js:22]: It should also be added to `system_user_manager` & `system_manager` roles

Comment (saturninoabril) [reply e2e/cypress/support/api/role.js:22]: I initially had those but were eventually removed after I've got the same bad request when patching the roles. See [logs](https://circleci.com/api/v1.1/project/github/saturninoabril/mattermost-cypress-docker/917/output/110/0?file=true&allocation-id=5f5486c12cdb5f297b4294e5-0-build%2F3F1CF14A). I've seen your recent PR for it. Maybe I'll wait for another build on master and see if it fixes the issue.

**Review by srkgupta — APPROVED** [review]
Ran the tests related to reset roles and they worked fine. The changes LGTM. Thanks @saturninoabril

**Comment by srkgupta** [comment]
Hi @saturninoabril, 
I am not approving the PR yet, as I still see the changes suggested by @fmunshi open. Please commit them and feel free to merge the PR.

**Comment by saturninoabril** [comment]
Thanks @srkgupta! Yup, I'll apply the comment but will wait for the new `master` docker, run tests against it and sort out final fix on master.

**Comment by saturninoabril** [comment]
@fmunshi Now added and looks good. Thanks for checking!

**Review by fmunshi — APPROVED** [review]

**Pull request merged**